torch/csrc/jit/mobile/interpreter.cpp

```diff
@@ -0,0 +1,124 @@
+#include "interpreter.h"
+#include <torch/csrc/jit/mobile/function.h>
+#include <aten/src/ATen/core/operator_name.h>
+#include <aten/src/ATen/core/dispatch/Dispatcher.h>
+
+namespace torch{
+namespace jit{
+char const * OpCode2Str(OpCode op);
+namespace mobile {
+InterpreterState::InterpreterState(std::shared_ptr<Code> code) : code_(code) {
+  registers_.resize(code_->register_size_);
+}
+
+//InterpreterState::InterpreterState(Function* function)
+//    : function_(function) {
+//  registers_.resize(function->register_size());
+//}
+
+bool InterpreterState::run(Stack& stack) {
+  size_t pc = 0;
+  while (true) {
+    //    std::cout << "RUNNING " << pc << " " << instructions_[pc];
+    //    std::cout << std::endl;
+    //    for (auto val : stack) {
+    //      if (val.isTensor()) {
+    //        std::cout << val.toTensor().sizes() << std::endl;
+    //      } else {
+    //        std::cout << val << std::endl;
+    //      }
+    //    }
+    Instruction inst = code_->instructions_[pc];
+    TORCH_CHECK(isOpSupportedInMobile(inst.op), OpCode2Str(inst.op),
+                " is not supported in mobile module.");
+    switch (inst.op) {
+      case OP: {
+        auto opname = code_->op_names_[inst.X];
+        auto op = c10::Dispatcher::singleton().findSchema(opname);
```

Comment:
This probably should not be done in the interpreter loop. The function pointer for the operator can be looked up and cached inside the Code object when the code object is created. That would also be a good time to do the name mangling necessary for looking up the right object.